packages/template-retail-react-app/app/components/display-price/current-price.jsx

```diff
@@ -17,42 +17,73 @@ import msg from '@salesforce/retail-react-app/app/components/display-price/messa
  * @param as - an HTML tag or component to be rendered as
  * @param isRange - show price as range or not
  * @param currency - currency to show the price in
+ * @param isLiveRegion - for a11y, a prop to indicate if this is a dynamical content
+ * @param name - product name
  * @param extraProps - extra props to be passed into Text Component
  * @returns {JSX.Element}
  */
-const CurrentPrice = ({price, as, isRange = false, currency, ...extraProps}) => {
+const CurrentPrice = ({
+    name,
+    isLiveRegion = false,
+    price,
+    as,
+    isRange = false,
+    currency,
+    ...extraProps
+}) => {
     const intl = useIntl()
     const currentPriceText = intl.formatNumber(price, {
         style: 'currency',
         currency
     })
-    return isRange ? (
-        <Text
-            as={as}
-            {...extraProps}
-            aria-label={intl.formatMessage(msg.ariaLabelCurrentPriceWithRange, {
-                currentPrice: currentPriceText
-            })}
-        >
-            {intl.formatMessage(msg.currentPriceWithRange, {
-                currentPrice: currentPriceText
-            })}
-        </Text>
-    ) : (
-        <Text
-            as={as}
-            {...extraProps}
-            aria-label={intl.formatMessage(msg.ariaLabelCurrentPrice, {
-                currentPrice: currentPriceText
-            })}
-        >
-            {currentPriceText}
-        </Text>
+    return (
+        <Box>
+            {isRange ? (
+                <Text
+                    as={as}
+                    {...extraProps}
+                    aria-label={intl.formatMessage(msg.ariaLabelCurrentPriceWithRange, {
+                        currentPrice: currentPriceText
+                    })}
+                >
+                    {intl.formatMessage(msg.currentPriceWithRange, {
+                        currentPrice: currentPriceText
+                    })}
+                </Text>
+            ) : (
+                <Text
+                    as={as}
+                    {...extraProps}
+                    aria-label={intl.formatMessage(msg.ariaLabelCurrentPrice, {
+                        currentPrice: currentPriceText
+                    })}
+                >
+                    {currentPriceText}
+                </Text>
+            )}
+            {/*For screen reader, we want to make sure the product name is announced before the price to avoid confusion*/}
+            <VisuallyHidden aria-live="polite" aria-atomic={true}>
```

Comment:
Do we need tests to ensure this element is rendered correctly?Verifying the aria-live, aria-atomic and content value?

Reply:
I don't really see the point since this is mainly for screen reader. We have to verify manually on the content to confirm it works. The tests may verify the text is there, but screen reader may not read it out loud